crates/dbsp/src/trace/ord/indexed_zset_batch.rs

```diff
@@ -390,6 +396,53 @@ where
     }
 }
 
+impl<K, V, R, O> Encode for OrdIndexedZSet<K, V, R, O>
+where
+    K: DBData,
+    V: DBData,
+    R: DBWeight,
+    O: OrdOffset,
+{
+    fn encode<E: Encoder>(&self, encoder: &mut E) -> Result<(), EncodeError> {
```

Comment:
`OrdZSet`, `OrdIndexedZSet`, and other `Batch` implementations are internally implemented as a bunch of vectors.  It might be much more efficient to encode these vectors directly rather than by iterating with the cursor API.

This is probably ok as the first cut, since we're experimenting anyway, but this simplest and also most efficient solution might be to auto-derive `Encode` and `Decode` for all types in `trace/layers`.

Reply:
OK, I'm working on that.

I guess that `Spine` needs to be encodable. It has an `Activator`, which has an `Instant` in it. That can't be portably encoded. How should `Spine` be encoded?

Reply:
I don't think the activator can be serialized.

Reply:
or that it should. I suspect you can just ignore it.

Reply:
I think that your current approach of making it a `todo!` is ok for now. We don't send spines across workers, only "normal" batches. As Mihai said, the activator doesn't need to be serialized.  Also, partially merged batches don't need to be serialized either. But let's just leave it a `todo!` until we actually need encodable batches.

Reply:
OK, great, thank you.

Reply:
I pushed a change for that. It does need to add `'static` bounds in a lot of places. I think that's a side effect of `Encode`.